src/plugins/DocManager/DocManager.js

```diff
@@ -141,7 +141,15 @@ class DocManager extends IDEPlugin {
    */
   discardDocChanges(modelKey) {
     const { name, scope } = modelKey;
-    return this.getStore(scope)?.discardDocChanges(name);
+
+    const ret = this.getStore(scope)?.discardDocChanges(name)
+
+    this.emit(PLUGINS.DOC_MANAGER.ON.DISCARD_DOC, {
+      name,
+      scope,
+    });
+
+    return ret;
```

Comment:
Any reason why the discard needs to be done before the emit?

```suggestion

    this.emit(PLUGINS.DOC_MANAGER.ON.DISCARD_DOC, {
      name,
      scope,
    });

    return this.getStore(scope)?.discardDocChanges(name);
```

Reply:
I guess this has to be tried out.

Reply:
Yes. There is a reason for it.